src/common_swagger_api/schema/data.clj

```diff
@@ -25,3 +33,75 @@
   (-> (merge DataIds OptionalPaths)
       st/optional-keys
       (describe "The path or data ids of the data objects to gather status information on.")))
+
+(def ValidFolderListingSortFields
+  #{:datecreated
+    :datemodified
+    :name
+    :path
+    :size})
+
+(s/defschema FolderListingPagingParams
+  (merge
+   PagingParams
+   {SortFieldOptionalKey
+    (describe (apply s/enum ValidFolderListingSortFields) SortFieldDocs)}))
+
+(s/defschema FolderListingParams
+  (merge
+   FolderListingPagingParams
+   {(s/optional-key :entity-type)
+    (describe (s/enum :any :file :folder) "The type of folder items to include in the response.")
+
+    (s/optional-key :info-type)
+    (describe (s/either [ValidInfoTypesEnum] ValidInfoTypesEnum)
+              "A list of info-types with which to filter a folder's result items.")}))
+
+(s/defschema PagedItemListEntry
+  {:infoType
+   (describe (s/maybe ValidInfoTypesEnum) "The info-type of the item being listed or null for folders")
+
+   :path
+   (describe String "The path to the item")
+
+   :date-created
+   (describe Long "The date and time that the item was created as a UNIX epoch")
+
+   :permission
+   (describe PermissionEnum "The user's permission-level for the item")
+
+   :date-modified
+   (describe Long "The date and time that the item was most recently modified as a UNIX epoch")
+
+   :file-size
+   (describe Long "The size of the item in bytes or 0 for folders")
+
+   :badName
+   (describe Boolean "True if the item name may cuase problems in the DE")
+
+   :isFavorite
+   (describe Boolean "True if the user has marked the item as a favorite")
+
+   :label
+   (describe String "The base name of the item")
+
+   :id
+   (describe UUID "The ID assigned to the item")})
+
+(s/defschema PagedFolderListing
+  (merge PagedItemListEntry
```

Comment:
It looks like some of these fields are specific to terrain, or are the terrain-renamed versions of these fields (`isFavorite`, `file-size` vs `size`, or `date-created` vs. `dateCreated`, for example).

I would expect that only the schemas common to both data-info and terrain would be added to this lib.

Unless we have plans to update `data-info` at some point to use the same response schema as terrain (which would be nice)?

Either way, there should probably be comments on these schemas stating that they are only compatible with `terrain` for now.

Reply:
Ah, good catch. I noticed that the endpoint in `data-info` has no response schema (because the response changes based on whether the path refers to a folder or file), so I created a schema based on what `terrain` returns without noticing that the response format in `data-info` is slightly different. What I might do is move this schema definition into `terrain` and call it good for now.

Reply:
OK, sounds good to me 👍